lib/membrane/rtp/outbound_rtx_controller.ex

```diff
@@ -51,49 +42,35 @@ defmodule Membrane.RTP.OutboundRtxController do
       ) do
     now = System.monotonic_time(:millisecond)
 
-    buffers_to_retransmit =
-      sequence_numbers
-      |> Stream.map(fn seq_num -> BufferStore.get_buffer(state.store, seq_num) end)
-      |> Stream.filter(fn
-        {:ok, buffer} ->
-          seq_num = buffer.metadata.rtp.sequence_number
-          last_rtx_time = Map.get(state.last_rtx_times, seq_num, now - @min_rtx_interval)
-
-          now - last_rtx_time >= @min_rtx_interval
-
-        {:error, :not_found} ->
-          false
+    {buffers, store} =
+      Enum.map_reduce(sequence_numbers, state.store, fn seq_num, store ->
+        maybe_retransmit(seq_num, now, store)
       end)
-      |> Enum.map(fn {:ok, buffer} -> buffer end)
 
-    state =
-      Map.update!(state, :last_rtx_times, fn times ->
-        updates =
-          Map.new(buffers_to_retransmit, fn buffer ->
-            {buffer.metadata.rtp.sequence_number, now}
-          end)
-
-        Map.merge(times, updates)
-      end)
+    buffers_to_retransmit = Enum.reject(buffers, fn buffer -> buffer == nil end)
 
     unless buffers_to_retransmit == [],
       do:
         Membrane.Logger.info(
-          "Retransmitting buffers with the following sequence numbers:\n#{inspect(Enum.map(buffers_to_retransmit, & &1.metadata.rtp.sequence_number))}"
+          "Retransmitting buffers (#{Enum.count(buffers_to_retransmit)}) with the following sequence numbers:\n#{inspect(Enum.map(buffers_to_retransmit, & &1.metadata.rtp.sequence_number))}"
         )
```

Comment:
I'd compact this log a bit: (cannot make suggestions due to deleted lines, GH sux)

```elixir
    unless buffers_to_retransmit == [] do
      Membrane.Logger.info("""
      Retransmitting (#{length(buffers_to_retransmit)}) buffers, sequence numbers:
      #{inspect(Enum.map(buffers_to_retransmit, & &1.metadata.rtp.sequence_number))}
      """)
    end
```